pkg/scheduler/scheduler_test.go

```diff
@@ -1488,3 +1488,80 @@ func TestUpdatePod(t *testing.T) {
 		})
 	}
 }
+
+func TestUpdatePodWithSecrets(t *testing.T) {
+	tests := []struct {
+		name           string
+		podName        string
+		secrets        []v1.LocalObjectReference
+		conditionToSet v1.PodCondition
+	}{
+		{
+			name:    "Should update pod conditions successfully with secrets with empty name",
+			podName: "pod1",
+			// this will serialize to imagePullSecrets:[{}]
+			secrets: make([]v1.LocalObjectReference, 1),
+			conditionToSet: v1.PodCondition{
+				Type:   v1.PodScheduled,
+				Status: v1.ConditionFalse,
+			},
+		},
+		{
+			name:    "Should update pod conditions successfully and preserve existing secrets",
+			podName: "pod2",
+			secrets: []v1.LocalObjectReference{{Name: "foo"}},
+			conditionToSet: v1.PodCondition{
+				Type:   v1.PodScheduled,
+				Status: v1.ConditionFalse,
+			},
+		},
+	}
+
+	client := clientsetfake.NewSimpleClientset()
+	ns := "ns"
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			pod := &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: ns,
+					Name:      tc.podName,
+				},
+				Status: v1.PodStatus{
+					Conditions: []v1.PodCondition{
+						{
+							Type:   v1.PodConditionType("abcd"),
+							Status: v1.ConditionTrue,
+						},
+					},
+				},
+			}
+			pod.Spec.ImagePullSecrets = tc.secrets
+			_, err := client.CoreV1().Pods(ns).Create(context.TODO(), pod, metav1.CreateOptions{})
+			if err != nil {
+				t.Fatal(err)
+			}
+			err = updatePod(client, pod, &tc.conditionToSet, "")
+			if err != nil {
+				t.Fatal(err)
+			}
+			retrievedPod, err := client.CoreV1().Pods(ns).Get(context.TODO(), tc.podName, metav1.GetOptions{})
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			if retrievedPod.Status.Conditions[0].Status != pod.Status.Conditions[0].Status || retrievedPod.Status.Conditions[0].Type != pod.Status.Conditions[0].Type {
+				t.Errorf("unexpected conditions set on pod: %v", retrievedPod.Status.Conditions)
+			}
+
+			if retrievedPod.Status.Conditions[1].Status != tc.conditionToSet.Status || retrievedPod.Status.Conditions[1].Type != tc.conditionToSet.Type {
+				t.Errorf("unexpected conditions set on pod: %v", retrievedPod.Status.Conditions)
+			}
+
+			if len(retrievedPod.Spec.ImagePullSecrets) > 0 {
```

Comment:
I think this check is too specific. For this bug, it was enough to check that updatePod doesn't return an error.

Reply:
yeah fair enough. Was mostly ensuring we don't wipe secrets but I agree not needed.

Reply:
@alculquicondor I moved the test to be under utils and test for `PatchPodStatus` instead since that's where the changed logic is now. lmk what you think.